File: PassTask/1.1P/program.py
# Function to calculate the average of an array
def average(numbers):
    total = 0  # Initialize sum variable
    count = len(numbers)  # Get the total number of elements

    for num in numbers:
        total += num

    # Return the average
    return total / count
# Step 5: Marshal data
data_values = [2.5, -1.4, -7.2, -11.7, -13.5, -13.5, -14.9, -15.2, -14.0, -9.7, -2.6, 2.1]

# Invoke the function
result = average(data_values)

student_name = "Show Wai Yan"
student_id = "105293041"

# Print results
print("Average of data values:", result)
print("Student Name:", student_name)
print("Student ID:", student_id)

# Business logic for interpreting the result
if result >= 10:
    print("Multiple digits")
else:
    print("Single digits")

if result < 0:
    print("Average value negative")

last_digit_avg = abs(int(result)) % 10  # Get last digit of absolute average value
last_digit_id = int(student_id[-1])  # Get last digit of student ID


if last_digit_avg > last_digit_id:
    print("Larger than my last digit")
elif last_digit_avg < last_digit_id:
    print("Smaller than my last digit")
else:
    print("Equal to my last digit")

File: PassTask/1.1P/Program.cs
using System;

class Program
{
    // Function to calculate the average of an array of integers
    static double Average(int[] numbers)
    {
        int sum = 0;
        int count = numbers.Length;

        // Loop through the array and calculate the sum
        for (int i = 0; i < count; i++)
        {
            sum += numbers[i];
        }

        // Return the average (sum divided by count)
        return (double)sum / count; // Cast to double for decimal precision
    }

    static void Main()
    {
        // Sample data values (converted to integers, since the function accepts int[])
        int[] data = { 2, -1, -7, -11, -13, -13, -14, -15, -14, -9, -2, 2 };

        // Call the function and store the result
        double avg = Average(data);

        // Print the result
        Console.WriteLine("The average of the data set is: " + avg);

        // Print student information
        Console.WriteLine("Student Name: Show Wai Yan");
        Console.WriteLine("Student ID: 12345678"); // Replace with actual ID

        // Business logic:
        // Check if the average is multiple digits or single digits
        if (avg >= 10)
        {
            Console.WriteLine("Multiple digits");
        }
        else
        {
            Console.WriteLine("Single digits");
        }

        // Check if the average is negative
        if (avg < 0)
        {
            Console.WriteLine("Average value negative");
        }

        // Check last digit of average vs last digit of student ID
        int lastDigitAvg = Math.Abs((int)avg) % 10; // Get last digit of the absolute average
        int lastDigitID = 8; // Last digit of Student ID (12345678)

        if (lastDigitAvg > lastDigitID)
        {
            Console.WriteLine("Larger than my last digit");
        }
        else if (lastDigitAvg < lastDigitID)
        {
            Console.WriteLine("Smaller than my last digit");
        }
        else
        {
            Console.WriteLine("Equal to my last digit");
        }
    }
}
